lib/mri/rubygems/specification_record.rb

```diff
@@ -15,7 +15,7 @@ def self.from_path(path)
     def initialize(dirs)
       @all = nil
       @stubs = nil
-      @stubs_by_name = {}
+      @stubs_by_name = TruffleRuby::ConcurrentMap.new
```

Comment:
We may not need this patch anymore after making Hash thread safe. And maybe `RubyConcurrentMap` class at all (or make it an alias to Hash)

Reply:
Yes indeed, I already reverted that patch in b29687abed47b86741fdb1965c3db0e0d65ee73c so it shouldn't be re-introduced.

Reply:
Re `TruffleRuby::ConcurrentMap` that needs to be kept for now because concurrent-ruby depends on it, it's also far more efficient than `Hash` concurrently because it doesn't need to preserve insertion ordering..

Reply:
I see, thank you!

Reply:
It looks like this file is identical to MRI 3.4.7 now, so I've modified the patch commit to remove these modifications.